test/test.js

```diff
@@ -29,6 +29,12 @@ describe('moo compiler', () => {
     expect(lexer.re.source.replace(/[(?:)]/g, '')).toBe('token|tok|t[ok]+|\\w')
   })
 
+  test('allows list of RegExps', () => {
+    expect(() => moo.compile({
+      tok: [/(foo)/, /(bar)/]
+    })).not.toThrow()
+  })
+
```

Comment:
You should add tests to ensure that the captured groups are returned correctly, even when they don't span the entire match.